usaspending_api/common/helpers/decorators.py

```diff
@@ -0,0 +1,40 @@
+import logging
+
+from django.db import connection
+from django.utils.decorators import method_decorator
+
+DEFAULT_DB_TIMEOUT_IN_MS = 300000
+logger = logging.getLogger(__name__)
+
+
+def set_db_timeout(timeout_in_ms=DEFAULT_DB_TIMEOUT_IN_MS):
+    """ Decorator used to set the database statement timeout within the Django app scope
+
+        Args:
+            timeout_in_ms: timeout value to set, in milliseconds
+
+        NOTE:
+            The statement_timeout is only set for this specific connection. Any subsequent request will have a default
```

Comment:
From my understanding, Django/uwsgi will maintain an idle connection to the DB as long as the worker exists. That would mean the setting does persist the specific API request handling.

Running `select * from pg_stat_activity;` in a production DB will typically show over 100 idle connections.

Reply:
I've added a reset